interop/src/test/scala/InteropTest.scala

```diff
@@ -19,19 +19,38 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import com.github.ghik.silencer.silent
+import cats.effect.{Resource, Sync}
+import log.effect.LogWriter
 import log.effect.interop.TestLogCapture
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpecLike
+import org.typelevel.log4cats.SelfAwareStructuredLogger
 
-@silent("local method [a-zA-Z0-9]+ in value <local InteropTest> is never used")
+import scala.annotation.nowarn
+
+sealed trait RedisClient[F[_]] {
+  def address: String
+  def write: F[Unit]
+}
+object RedisClient {
+  def apply[F[_]: LogWriter](addr: String)(implicit F: Sync[F]): Resource[F, RedisClient[F]] =
+    Resource.make(
+      F.pure(
+        new RedisClient[F] {
+          val address        = addr
+          def write: F[Unit] = LogWriter.info(address)
+        }
+      )
+    )(_ => F.unit)
+}
```

Comment:
The same issue with inlining is avoided here by extracting the sealed trait and Object outside of the tests